engine/src/main/java/org/terasology/config/flexible/FlexibleConfig.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.config.flexible;
+
+import com.google.common.collect.Maps;
+import org.terasology.assets.ResourceUrn;
+
+import java.util.Map;
+
+public class FlexibleConfig {
+    private Map<Key<?>, Setting> settingMap;
+
+    public FlexibleConfig() {
+        this.settingMap = Maps.newHashMap();
+    }
+
+    public <V> Key<V> add(Setting<V> setting) {
+        Key<V> key = new Key<>(setting.getId(), setting.getValue());
+
+        // Maybe throw an exception?
+        if (has(key))
+            return null;
```

Comment:
I return the `Key` because the `Key` is created by the `FlexibleConfig` itself, and needs to be used to access the setting from `FlexibleConfig.get`. Useful in cases like:

```
Key<Float> key = flexibleConfig.add(someFloatSetting);
// ...
Setting<Float> setting = flexibleConfig.get(key);
```

Of course, if keys are going to be manually created (like in a static helper class), I can create an overload of `add`: `boolean add(Key<V>, Setting<V>)` that returns true or false showing whether the `Key` was added. What do you think I should do with the existing `add(Setting)` method?